python/pyspark/cloudpickle.py

```diff
@@ -326,7 +329,10 @@ def save_global(self, obj, name=None, pack=struct.pack):
 
         modname = getattr(obj, "__module__", None)
         if modname is None:
-            modname = pickle.whichmodule(obj, name)
+            try:
+                modname = pickle.whichmodule(obj, name)
+            except Exception:
+                modname = '__main__'
```

Comment:
The use of `__main__` here seems consistent with the error-handling path on line 181 👍